### PR TITLE
Add script to generate user permissions file from google spreadsheets

### PR DESCRIPTION
**Pull request by vmahedia**

### CODE REVIEW

File: scripts/generate_user_permissions.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import os 
+import argparse
+import gspread
+from oauth2client.service_account import ServiceAccountCredentials
+
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-f", "--oauth_credentials_file", help="Path of the file where oauth credetials to " +
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: minor typo - *credentials* - in help.

File: scripts/generate_user_permissions.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    "access google drive API is stored, Make sure the email in this file " + 
+                    "has edit access on the spreadsheets being read in this script")
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: is edit access still required?

File: scripts/generate_user_permissions.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    parser.add_argument('-ss', "--spreadsheet_title", action='append')
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: good!! :+1:

File: scripts/generate_user_permissions.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    parser.add_argument('-p', '--password_column_name', help="Name of the column in the spreadsheet which has password,"+
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: minor - space after comma - `password,`
question - Why not make this default to "username" or even make it mandatory so there is not explicit default logic in code?

File: scripts/generate_user_permissions.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    actions = ["create", "update", "delete", "view", "killTask"]
+    permissions_json = {}
+    credentials = {}
+
+    for row in rows:
+        # todo : "username" and "path" are required keys, bailout if they are not defined
+        # dictionary with keys username, team, path, create, update, delete, view, killTask
+        actions_allowed = []
+        user = row["username"]
+        if (row["path"] == None or row["path"] == ""):
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: can be replaced with `if not row["path"]`

File: scripts/generate_user_permissions.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+        for action in actions_allowed:    
+            permission = {}
+            permission["allowed"] = action
+            permission["on"] = row["path"]
+            if (user not in permissions_json.keys()):
+                permissions_json[user] = []
+            permissions_json[user].append(permission) 
+
+        if (credentials.get(user) == None or credentials.get(user) == ""):
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: `if credentials.get(user):`

File: scripts/generate_user_permissions.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            if (password_column in row.keys()):
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: no need for this check if `password_column` is mandatory (so there's no explicit logic in code for defaults.)

File: scripts/generate_user_permissions.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+                credentials[user] = row[password_column]
+            else:
+                credentials[user] = user
+                
+    for user, password in credentials.iteritems():
+        user_permissions_json.append ({"user": user.lower(), "password": password.lower(), "permissions": permissions_json[user]})
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]: `password` should not be changed to lower.

File: scripts/generate_user_permissions.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+    users_json = {"users": user_permissions_json}
+    return users_json
+
+def writeJsonToFile(json_to_write, filename):
+    with open(filename, 'w') as outfile:
+        json.dump(json_to_write, outfile, indent=4, sort_keys=True)
+    print "Successfully generated Json file - %s" % os.getcwd()  + "/" + filename
+                
+main()
```
Comment (ankanm) [inline scripts/generate_user_permissions.py]:
```
if __name__ == "__main__":
    main()
```

File: scripts/generate_user_permissions.py (hunk 2 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: Nope.

File: scripts/generate_user_permissions.py (hunk 4 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: +1

File: scripts/generate_user_permissions.py (hunk 5 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: :+1:

File: scripts/generate_user_permissions.py (hunk 8 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: :+1:  very good catch.

File: scripts/generate_user_permissions.py (hunk 9 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: :+1:

File: scripts/generate_user_permissions.py (hunk 1 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: :white_check_mark:

File: scripts/generate_user_permissions.py (hunk 2 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: :white_check_mark:

File: scripts/generate_user_permissions.py (hunk 6 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: :white_check_mark:

File: scripts/generate_user_permissions.py (hunk 7 above)
Comment (vmahedia) [inline scripts/generate_user_permissions.py]: made it optional but defaulting to username so removed the check 

:white_check_mark:

**Comment by vmahedia** [comment]
@ankanm

**Comment by ankanm** [comment]
LGTM

**Pull request merged**